token-lending/program/src/processor.rs

```diff
@@ -1769,6 +1945,18 @@ fn get_pyth_product_quote_currency(pyth_product: &pyth::Product) -> Result<[u8;
     Err(LendingError::InvalidOracleConfig.into())
 }
 
+fn get_price(
+    switchboard_feed_info: &AccountInfo,
+    pyth_price_account_info: &AccountInfo,
+    clock: &Clock,
+) -> Result<Decimal, ProgramError> {
+    let pyth_price = get_pyth_price(pyth_price_account_info, clock).unwrap_or_default();
+    if pyth_price != Decimal::zero() {
```

Comment:
What primarily causes this to occur such that it fails over to Switchboard for the price? Is it staleness of the Pyth price?

Reply:
usually staleness yes, we have had several instances of this happening on devnet, a bunch of feeds were down for 12 hours once and one feed was down for ~3 days